From f2a10d4674bcfae006fa3c4ed0846addd72b120b Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Thu, 23 Oct 2014 09:54:55 -0400
Subject: [fs] ext4: fix 64-bit number truncation warning

Message-id: <1414058125-4183-142-git-send-email-lczerner@redhat.com>
Patchwork-id: 98241
O-Subject: [RHEL 7.1 PATCH v2 141/171] ext4: fix 64-bit number truncation warning
Bugzilla: 1150139
RH-Acked-by: Eric Sandeen <sandeen@redhat.com>

From: Chen Gang <gang.chen.5i5j@gmail.com>

BZ 1150139
https://bugzilla.redhat.com/show_bug.cgi?id=1150139

BREW 8142710
https://brewweb.devel.redhat.com/taskinfo?taskID=8142710

Upstream commit 666525dfbdca09bbd4848ac711e4a4dbd6921325

 '0x7FDEADBEEF' will be truncated to 32-bit number under unicore32. Need
 append 'ULL' for it.

 The related warning (with allmodconfig under unicore32):

     CC [M]  fs/ext4/extents_status.o
   fs/ext4/extents_status.c: In function "__es_remove_extent":
   fs/ext4/extents_status.c:813: warning: integer constant is too large for "long" type

 Signed-off-by: Chen Gang <gang.chen.5i5j@gmail.com>
 Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>
 Signed-off-by: Lukas Czerner <lczerner@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/extents_status.c b/fs/ext4/extents_status.c
index 6f98747..4567602 100644
--- a/fs/ext4/extents_status.c
+++ b/fs/ext4/extents_status.c
@@ -810,7 +810,7 @@ retry:
 
    newes.es_lblk = end + 1;
    newes.es_len = len2;
-   block = 0x7FDEADBEEF;
+   block = 0x7FDEADBEEFULL;
    if (ext4_es_is_written(&orig_es) ||
        ext4_es_is_unwritten(&orig_es))
     block = ext4_es_pblock(&orig_es) +
-- 
1.7.1